### PR TITLE
CLI: Volume Expand option for gd2 CLI

### PR DESCRIPTION
**Pull request by devyanikota**
Updates #205: Add volume expand option to expand a Gluster Volume
by adding more bricks.

Signed-off-by: Devyani Kota <devyanikota@gmail.com>

### CODE REVIEW

**Comment by devyanikota** [comment]
@aravindavk am able to add more bricks to (expand) an existing volume.
Can you please review the same? Thank you.

File: cli/cmd/volume.go [hunk 1]
```diff
@@ -73,6 +74,7 @@ func init() {
 	volumeCmd.AddCommand(volumeInfoCmd)
 	volumeCmd.AddCommand(volumeStatusCmd)
 	volumeCmd.AddCommand(volumeListCmd)
+	volumeCmd.AddCommand(volumeExpandCmd)
```
Comment (aravindavk) [inline cli/cmd/volume.go]: Before this line, Flag need to be defined

	volumeExpandCmd.Flags().IntVarP(&flagCreateCmdReplicaCount, "replica", "", 0, "Replica Count")

File: cli/cmd/volume.go [hunk 2]
```diff
@@ -325,3 +327,27 @@ var volumeStatusCmd = &cobra.Command{
 		fmt.Println("STATUS:", volname)
 	},
 }
+
+var volumeExpandCmd = &cobra.Command{
+	Use:   "expand",
```
Comment (aravindavk) [inline cli/cmd/volume.go]: Change this to "add-brick" instead of "expand" for backward compatibility.

File: cli/cmd/volume.go [hunk 3]
```diff
@@ -325,3 +327,27 @@ var volumeStatusCmd = &cobra.Command{
...
+	Short: helpVolumeExpandCmd,
+	Run: func(cmd *cobra.Command, args []string) {
+		validateNArgs(cmd, 2, 0)
+		volname := cmd.Flags().Args()[0]
+		bricks, err := bricksAsUUID(cmd.Flags().Args()[1:])
+		if err != nil {
+			log.WithField("volume", volname).Println("volume expansion failed")
```
Comment (aravindavk) [inline cli/cmd/volume.go]: Same as above, change all messages to "add-brick"

Comment (devyanikota) [reply cli/cmd/volume.go]: Worked on the suggested changes. Thanks.

File: pkg/api/req.go [hunk 4]
```diff
@@ -19,3 +19,11 @@ type PeerAddReq struct {
 type VolOptionReq struct {
 	Options map[string]string `json:"options"`
 }
+
+// VolExpandReq represents a request to expand the volume by adding more bricks
+type VolExpandReq struct {
```
Comment (prashanthpai) [inline pkg/api/req.go]: Also, please remove the copy of this struct from `commands/volumes/volume-expand.go`. Import it there.

File: cli/cmd/volume.go [hunk 5]
```diff
@@ -325,3 +331,27 @@ var volumeStatusCmd = &cobra.Command{
...
+		vol, err := client.VolumeExpand(volname, api.VolExpandReq{
+			ReplicaCount: flagCreateCmdReplicaCount,
+			Bricks:       bricks, // string of format <UUID>:<path>
+		})
+		if err != nil {
+			log.WithField("volume", volname).Println("volume expansion failed")
+			failure(fmt.Sprintf("Addition of brick failed with %s", err.Error()), 1)
+		}
+		fmt.Printf("%s Volume expanded successfully\n", vol.Name)
+		fmt.Println("Volume ID: ", vol.ID)
```
Comment (prashanthpai) [inline cli/cmd/volume.go]: Printing volume id is useful only during volume create. You can remove this line.

**Comment by devyanikota** [comment]
@aravindavk @prashanthpai updated the PR with the suggested changes, please review the same. Thanks.

**Comment by prashanthpai** [comment]
@devyanikota Please ensure that the tests pass.

**Comment by devyanikota** [comment]
@prashanthpai nod nod. Edited the tests accordingly.

**Comment by devyanikota** [comment]
@prashanthpai removed reqapi, imported directly.

**Pull request merged**